sui/src/rpc_gateway.rs

```diff
@@ -53,24 +68,28 @@ pub trait RpcGateway {
         recipient: SuiAddress,
     ) -> RpcResult<TransactionBytes>;
 
+    /// Execute a Move call transaction by calling the specified function in the module of a given package.
     #[method(name = "moveCall")]
     async fn move_call(
         &self,
         signer: SuiAddress,
         package_object_id: ObjectID,
-        module: Identifier,
-        function: Identifier,
-        type_arguments: Vec<TypeTag>,
+        #[schemars(with = "json_schema::Identifier")] module: Identifier,
+        #[schemars(with = "json_schema::Identifier")] function: Identifier,
+        #[schemars(with = "Option<Vec<json_schema::TypeTag>>")] type_arguments: Option<
+            Vec<TypeTag>,
+        >,
         arguments: Vec<RpcCallArg>,
         gas_object_id: ObjectID,
         gas_budget: u64,
```

Comment:
N00b question: why do we use the schemar marco on line 77 but not on line 82?

Reply:
the `schemars` tag is used to tell the macro to generate schema with the provided Type instead of the actual declared Type, we need this because `Identifier` and `TypeTag` belongs to the move package so we can't implement `JsonSchema` trait for them, instead we use proxy types in `json_schema` to generate the schema.